tests/cpp/test_host_ir_integration.cpp

```diff
@@ -177,6 +181,50 @@ TEST_F(HostIrIntegrationTest, Deallocate) {
   EXPECT_EQ(memoryAllocated(device_index), 0);
 }
 
+TEST_F(HostIrIntegrationTest, InsertDeallocations) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+  TensorView* in = makeSymbolicTensor(2);
+  fusion->addInput(in);
+
+  TensorView* t0 = add(in, in);
+  auto t1 = segment_set(t0);
+  TensorView* t2 = add(t1, t1);
+  auto t3 = segment_set(t2);
+  TensorView* t4 = add(t3, t0);
+  TensorView* out = add(t4, t4);
+  fusion->addOutput(out);
+
+  EnableOptionsGuard opt_guard;
+  EnableOptionsGuard::getCurOptions().set(EnableOption::HostIrLowering);
+
+  FusionExecutorCache executor_cache(std::move(fusion));
+  at::Tensor in_tensor =
+      at::randn({2, 3}, at::dtype(at::kFloat).device(at::kCUDA, 0));
+  auto out_tensors = executor_cache.runFusionWithInputs({in_tensor});
+
+  FusionKernelRuntime* runtime = executor_cache.getMostRecentKernelRuntime();
+  EXPECT_EQ(runtime->getHostIrEvaluator().canRun(), "");
+  auto hicExprs =
+      runtime->getHostIrEvaluator().getHostIrContainer().topLevelExprs();
+  using testing::Contains;
+  using testing::IsTrue;
+  using testing::Property;
```

Comment:
Move these to the beginning of the namespace. They can and tend to be used by other tests as well.